src/popup/router/pages/SignAndVerifyMsg.vue

```diff
@@ -82,43 +85,40 @@ export default {
     },
     methods:{
         signMessageAction() {
-                this.loading = true;
-                browser.storage.sync.get('userAccount').then(async (user) => {
-                    if(user.userAccount && user.hasOwnProperty('userAccount')) {
-                        let encPrivateKey = user.userAccount.encryptedPrivateKey;
-                        try {
-                            JSON.parse(user.userAccount.encryptedPrivateKey);
-                        }catch(e) {
-                            user.userAccount.encryptedPrivateKey = JSON.stringify( user.userAccount.encryptedPrivateKey );
-                            encPrivateKey = JSON.stringify( user.userAccount.encryptedPrivateKey );
-                        }
-                        let encryptedPrivateKey = JSON.parse(user.userAccount.encryptedPrivateKey);
-                        let privKey = await decrypt(encryptedPrivateKey.crypto.ciphertext,'123123123',encryptedPrivateKey.crypto.cipher_params.nonce,encryptedPrivateKey.crypto.kdf_params.salt);
-                        let privKey64 = Buffer.from(privKey, 'hex')
-                        // let privKey64 = str2buf(privKey);
-                        try {
-                            let sign = Crypto.signPersonalMessage(this.signMessage, privKey64)
-                            this.signature = JSON.stringify(
-                                {
-                                    // msg: Crypto.hash(this.signMessage),
-                                    text: this.signMessage,
-                                    sig: Buffer.from(sign),
-                                    address: this.account.publicKey,
-                                },
-                                null,
-                                2
-                            );
-                            this.loading = false;
-                            this.modalVisible = true;
-                        } catch (error) {
-                            console.log(error);
-                        }
+            this.loading = true;
+            browser.storage.sync.get('userAccount').then(async (user) => {
+                if(user.userAccount && user.hasOwnProperty('userAccount')) {
+                    let encPrivateKey = user.userAccount.encryptedPrivateKey;
+                    try {
+                        JSON.parse(user.userAccount.encryptedPrivateKey);
+                    }catch(e) {
+                        user.userAccount.encryptedPrivateKey = JSON.stringify( user.userAccount.encryptedPrivateKey );
+                        encPrivateKey = JSON.stringify( user.userAccount.encryptedPrivateKey );
                     }
-                });
+                    let encryptedPrivateKey = JSON.parse(user.userAccount.encryptedPrivateKey);
+                    let privKeyasHex = await decrypt(encryptedPrivateKey.crypto.ciphertext,'123123123',encryptedPrivateKey.crypto.cipher_params.nonce,encryptedPrivateKey.crypto.kdf_params.salt);
```

Comment:
Please, add a popup/prompt for password here, so the user can enter it when they signs a message. And get rid of the hardcoded `123123123`.

Reply:
It will be done but anyway the functionality for sign/verify message is still not working.